### PR TITLE
library: Ported `Switch` to RUST

### PR DESCRIPTION
**Pull request by onkarrai06**
Read the JS code and tried to convert it to RUST. However, this has an ownership error that I can't seem to resolve as Switch does not implement Copy. Would appreciate any help as I've spent a long time on this.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
   "rust-analyzer.runnables.command": "${workspaceFolder}/.flatpak/cargo.sh",
   "rust-analyzer.files.excludeDirs": [".flatpak"],
   "[yaml]": {
     "editor.defaultFormatter": "esbenp.prettier-vscode"
-  }
+  },
+  "mesonbuild.configureOnOpen": false,
+  "mesonbuild.buildFolder": "_build",
+  "mesonbuild.mesonPath": "${workspaceFolder}/.flatpak/meson.sh"
 }
```
Comment (Hofer-Julian) [inline .vscode/settings.json:33]: Please remove this change :)

File: src/Library/demos/Switch/code.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+pub fn main() {
+    let switch_on: gtk::Switch = workbench::builder().object("switch_on").unwrap();
+    let label_on: gtk::Label = workbench::builder().object("label_on").unwrap();
+
+    let switch_off: gtk::Switch = workbench::builder().object("switch_off").unwrap();
+    let label_off: gtk::Label = workbench::builder().object("label_off").unwrap();
+
+    let switch_on_clone = switch_on.clone();
+    let switch_off_clone = switch_off.clone();
```
Comment (Hofer-Julian) [inline src/Library/demos/Switch/code.rs:12]: Make weak references with `glib::clone` instead.
This should solve your lifetime issues.

You can read more about this here: https://gtk-rs.org/gtk4-rs/stable/latest/book/g_object_memory_management.html

**Review by Hofer-Julian — CHANGES_REQUESTED** [review]

File: src/Library/demos/Switch/code.rs (hunk 2 above)
Comment (onkarrai06) [reply src/Library/demos/Switch/code.rs]: Changed, please check now.

File: src/Library/demos/Switch/code.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+use gtk::prelude::*;
+
+pub fn main() {
+    let switch_on: gtk::Switch = workbench::builder().object("switch_on").unwrap();
+    let label_on: gtk::Label = workbench::builder().object("label_on").unwrap();
+
+    let switch_off: gtk::Switch = workbench::builder().object("switch_off").unwrap();
+    let label_off: gtk::Label = workbench::builder().object("label_off").unwrap();
+
+    switch_on.connect_active_notify(clone!(@weak switch_off, @weak switch_on => move |_| {
```
Comment (saadulkh) [inline src/Library/demos/Switch/code.rs:13]:
```suggestion
    switch_on.connect_active_notify(clone!(@weak switch_off => move |switch_on| {
```

File: src/Library/demos/Switch/code.rs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        label_on.set_label(if switch_on.is_active() {
+            "On"
+        } else {
+            "Off"
+        });
+        switch_off.set_active(!switch_on.is_active());
+    }));
+
+    switch_off.connect_active_notify(clone!(@weak switch_off, @weak switch_on => move |_| {
```
Comment (saadulkh) [inline src/Library/demos/Switch/code.rs:22]:
```suggestion
    switch_off.connect_active_notify(move |switch_off| {
```

File: src/Library/demos/Switch/code.rs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    let switch_off: gtk::Switch = workbench::builder().object("switch_off").unwrap();
+    let label_off: gtk::Label = workbench::builder().object("label_off").unwrap();
+
+    switch_on.connect_active_notify(clone!(@weak switch_off, @weak switch_on => move |_| {
+        label_on.set_label(if switch_on.is_active() {
+            "On"
+        } else {
+            "Off"
+        });
```
Comment (saadulkh) [inline src/Library/demos/Switch/code.rs:18]:
```suggestion
        label_on.set_label(if switch_on.is_active() { "On" } else { "Off" });
```

**Review by saadulkh — CHANGES_REQUESTED** [review]
Looks good to me otherwise. @Hofer-Julian Have a look. :)

**Comment by onkarrai06** [comment]
The suggestions above work on my side, waiting on final review :)

**Review by Hofer-Julian — APPROVED** [review]
The code looks good to me.
If @saadulkh can confirm that it works as expected, I will merge it.
Otherwise, I'll check tomorrow.

**Comment by saadulkh** [comment]
Tried code from the latest commit https://github.com/sonnyp/Workbench/pull/561/commits/2980d94cc97b90b28ef6d57e9d97935c21489296
Works as expected. @Hofer-Julian 

Thanks for the addition. @onkarrai06

**Pull request merged**